src/gui/qgisinterface.h

```diff
@@ -1280,6 +1280,16 @@ class GUI_EXPORT QgisInterface : public QObject
     */
     virtual void unregisterDevToolWidgetFactory( QgsDevToolWidgetFactory *factory ) = 0;
 
+    /**
+     * Show a page of the API documentation
+     * \param api "pyqgis" or "qgis" or "qt" or "pyqgis-search"
+     * \param embedded If TRUE, the documentation will be opened in the embedded devtools webview. Otherwise, use system web browser
+     * \param object object to show in the documentation
+     * \param module used only if api = "pyqgis"
+     * \since QGIS 3.42
+    */
+    virtual void showApiDocumentation( const QString &api = QStringLiteral( "pyqgis" ), bool embedded = true, const QString &object = QString(), const QString &module = QString() ) = 0;
```

Comment:
Instead of a raw string for `api`, can you add an enum class into the Qgis class for this? 

I would also change the boolean "embedded" to an explicit enum class too, eg:

```
enum class DocumentationBrowser
{
   DeveloperToolsPanel,
   SystemWebBrowser
}
```

This makes the code more readable (and slightly more future proof) vs a raw boolean